pkg/multiboot/module.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Module defines modules to be loaded along with the kernel.
+// https://www.gnu.org/software/grub/manual/multiboot/multiboot.html#Boot-information-format.
+
+package multiboot
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/ubinary"
+)
+
+var errEmptyModule = errors.New("empty module")
+
+// A Module represents a module to be loaded along with the kernel.
+type Module struct {
+	// Start is the inclusive start of the Module memory location
+	Start uint32
+	// End is the exclusive end of the Module memory location.
+	End uint32
+
+	// CmdLine is a pointer to a null-terminated ASCII string.
+	CmdLine uint32
+
+	// Reserved is always zero.
+	Reserved uint32
+}
+
+type module struct {
```

Comment:
can we distinguish the names of Module and module somehow?

Reply:
Module -> LoadedModule